rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/WorkerBookKeeper.java

```diff
@@ -176,10 +176,10 @@ private void initializeNodeStateCache(final Configuration conf, final Ticker tic
               return client.getClusterNodes();
             }
             catch (Exception e) {
-              client.close();
               throw e;
             }
             finally {
+              bookKeeperFactory.returnBookKeeperClient(client.getTransportPoolable());
```

Comment:
Instead of each user of BookKeeperClient to be aware of pooling keep the pooling work limited to BookKeeperClient. Continue to have BookKeeperClient.close() which internally releases the transport.

Reply:
moved logic in close method and user try with auto closeable wherever needed.